backend/estimate-service/src/main/java/org/egov/works/service/WorkflowService.java

```diff
@@ -0,0 +1,254 @@
+package org.egov.works.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import digit.models.coremodels.*;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.common.contract.request.User;
+import org.egov.tracer.model.CustomException;
+import org.egov.works.config.EstimateServiceConfiguration;
+import org.egov.works.repository.ServiceRequestRepository;
+import org.egov.works.web.models.Estimate;
+import org.egov.works.web.models.EstimateRequest;
+import org.egov.works.web.models.EstimateRequestWorkflow;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.egov.works.util.EstimateServiceConstant.ESTIMATE_BUSINESSSERVICE;
+import static org.egov.works.util.EstimateServiceConstant.ESTIMATE_MODULE_NAME;
+
+@Service
+public class WorkflowService {
+
+    @Autowired
+    private EstimateServiceConfiguration serviceConfiguration;
+
+    @Autowired
+    private ServiceRequestRepository repository;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+
+    /*
+     * Should return the applicable BusinessService for the given request
+     *
+     */
+    public BusinessService getBusinessService(EstimateRequest estimateRequest) {
+        String tenantId = estimateRequest.getEstimate().getTenantId();
+        String rootTenantId = tenantId.split("\\.")[0];
+        StringBuilder url = getSearchURLWithParams(rootTenantId, ESTIMATE_BUSINESSSERVICE);
```

Comment:
Can we avoid overhead of this api call as the transition api call returns with the same error message if the business service does not exist? 
"No bussinessService object found for businessSerice: loi-approval-5 and tenantId: pb"

Reply:
I have override the message as "The businessService : estimate-approval-2 doesn't exist" with message code as  "BUSINESSSERVICE_DOESN'T_EXIST"